fms/modules/attention.py

```diff
@@ -83,28 +87,25 @@ def __init__(
 
     def reset_params(self, gain=1):
         # Ensure softmax inputs are standard normal
-        for layer in ["query", "key"]:
-            nn.init.trunc_normal_(
-                getattr(self, layer).weight, mean=0.0, std=self.emb_dim**-0.5
-            )
+        self.qkv_fused.weight.data[
+            : self.emb_kq_per_head * (self.nheads + self.kvheads)
+        ].normal_(0, self.emb_dim**-0.5)
         # Ensure projection layers have same scale (for normalized-step dataloaders like
         # AdamW / Sophia), and maintain input norm up to attention remix, in expectation
-        for layer in ["value", "dense"]:
-            nn.init.trunc_normal_(
-                getattr(self, layer).weight,
-                mean=0.0,
-                std=(gain / (self.emb_dim * self.nheads * self.emb_v_per_head) ** 0.5)
-                ** 0.5,
-            )  # Using explicit terms instead of numel to account for eventual MQA addition
+        self.qkv_fused.weight.data[
```

Comment:
typo here? instead of qkv_fused shouldn't it be the dense matrix?

Reply:
yes, this needs to be fixed, going to fix in latest commit. One thing to note, this will most likely be replaced in a newer version as the reset params is being updated in another pr